components/tikv_util/src/config.rs

```diff
@@ -398,7 +398,7 @@ impl<'de> Deserialize<'de> for ReadableDuration {
                         ));
                     }
                     // do we need to check 12h360m?
-                    let number_str = unsafe { str::from_utf8_unchecked(first) };
+                    let number_str = str::from_utf8(first).expect("utf8");
```

Comment:
But apparently first has to be encoded in utf-8, why add an unnecessary check?

Reply:
Performance doesn't matter here. We shouldn't be writing unsafe code unless necessary.

Reply:
What's the harm when we know for sure it's safe to write an unsafe?

Reply:
People too often do not know for sure that it is safe to write unsafe, even if they think they know. By not writing 'unsafe' at all it is obviously not unsafe. Every appearance of unsafe requires that some invariant must hold, not just now, but in the future as other developers maintain that unsafe code.

Reply:
FWIW, this is just one case of unnecessary unsafe, and whether or not it is removed doesn't matter much. I think the TiKV codebase has much too much unsafe code though.

Reply:
For now I've removed the commit that removes the use of `from_utf8_unchecked`. I may return to this issue more generally in the future.